apps/ftu/js/late_customization.js

```diff
@@ -0,0 +1,657 @@
+          if (operatorInfos.operator) {
+            operatorInfos.mcc = conn.voice.network.mcc;
+            operatorInfos.mnc = conn.voice.network.mnc;
+            return operatorInfos;
+          }
+        }
+      }
+      return null;
+    },
+
+    getAppsFromManifest: function(data) {
+      var apps = new Map();
+      var langs = navigator.languages;
+      (data.apps || data.objects).filter((app) => {
+        return app.is_disabled !== true &&
+               (new Set(app.device_types)).has('firefoxos');
+      }).forEach((appData) => {
+        appData.manifestURL = appData.manifest_url;
+        // get locale-appropriate name for user display
+        var appName = '';
+        langs.some(function tryLanguage(lang) {
+          if (this[lang]) {
+            appName = this[lang];
+            // aborts [].some
+            return true;
+          }
+        }, appData.name);
+
+        appData.displayName = appName;
+        this.debug('app data: ', appData.displayName, appData);
+        apps.set(appData.manifestURL, appData);
+      });
+      return apps;
+    },
+
+    attemptInstalls: function() {
+      this._installQueue = [];
+      if (this._appsToInstall && this._appsToInstall.size) {
+        this._appsToInstall.forEach(appData => {
+          if (appData.installState !== 'installed' ||
+              appData.installState !== 'pending') {
+            this.enqueAppInstall(appData);
+          }
+        });
+      }
+    },
+    installApp: function(appData) {
+      this.debug('installApp: ', appData.manifestURL);
+      return this.whenUsableConnection().then(() => {
+        return new Promise((resolve, reject) => {
+          var req = window.navigator.mozApps.install(appData.manifestURL);
+          var onOutcome = (arg) => {
+            var selector = `li[data-manifest-url="${appData.manifestURL}"]`;
+            var itemNode = this.appListNode.querySelector(selector);
+            var manifestURL = req.result && req.result.manifestURL;
+
+            this.debug('appinstall DOMRequest outcome:', arg, req.result);
+            if (req.error) {
+              delete appData.installState;
+              itemNode && itemNode.classList.add('failed');
+              console.warn('Install error: ', req.error);
+            } else if (req.result) {
+              appData.installState = req.result.installState;
+              this.debug('Install success, app: ',
+                          manifestURL, appData.installState);
+            }
+            itemNode && itemNode.classList.remove('pending');
+            resolve(req);
+          };
+          req.onsuccess = req.onerror = onOutcome;
+        });
+      }).catch(() => {
+          this.debug('Lost connection while attempting install of ' +
+                    appData.manifestURL);
+      });
+    },
+
+    enqueAppInstall: function(app) {
+      var installQueue = this._installQueue || (this._installQueue = []);
```

Comment:
nit:
we already initialized  `this._installQueue` in `attemptInstalls` no need to do it again.